### ISSUE TITLE
Item Object

### ISSUE THREAD

**Issue by gslance** [jondeaves/tcm-bot#23]
A character should be able to use an item. An item can be any of the following sub-types:

- Consumable Items, e.g health potions
- Equippable Items, e.g swords
- Effect Items, e.g magic scrolls
- Currency Items, e.g coin

An item object would consist of the following data:

- [ ] Display name (string)
- [ ] Item Type (see above) (enumerator)
- [ ] Value (int)
- [ ] Power (int)
- [ ] Defense (int)
- [ ] Magic (int)
- [ ] Item Effect  (array)
- [ ] Flavor (string)
- [ ] Use Message (string)

**Value**
The numeric value may be used to note the importance or rarity of this given item. It may be used to trade an item for an equal amount of currency items (which are each worth one). Currency items may in turn be traded for other items.

**Power, Defense and Magic**
The base statistics of each item may be used depending on the type of the item.

The Power value is used to determine by how much a type of effect is doing for. In healing effects, the Power may be used to increase health by that amount of its target, while in damage effect the same number could be used to reduce ones health.

Defense and Magic, while useless for consumables, in addition to Power the statistics would be used to add ontop of the character were this item equipped or bestowed in some way. While the item is marked as equipped, these statistics would be added to its user.

**Item Effect**
This effect would be an enumerator of effects capable of impacting its user. Example effects are:

- `HEAL_HP`
- `REMOVE_POISON`
- `ADD_INVISIBLE`
- `DAMAGE_HP`
- `ADD_STUN`
- `REMOVE_REGENERATE`

A magic scroll or other Effect item, would cast this effect on a target instead.

**Flavor**
The flavor text may be used to describe an item if a user looks at it. This may be generated by reading what the item type is and its effect but it would be beneficial for certain types to hand write these (e.g quest items).

**Use Message**
An item being consumed, equipped or used in any way would be capable of producing a message of its use. This may be generated by its type and effect but could also be hand written.

**Comment by chenasraf** [comment]
Should we consider `quantity` property or similar? Imagine stacking 100 arrows as array duplicates... seems like it could cause issues down the road. What do you think?

**Comment by gslance** [comment]
You're right. Initially I was going to just allow duplicates to live in the array, but what you suggest is a cleaner approach. However, at this time I'm not entirely sure how items will work in this casual game yet. We need to discuss what form they should have, if at all.
